distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -410,14 +278,19 @@ atom_with_hole factory = @Builtin_Method "Meta.atom_with_hole_builtin"
 
    Arguments:
    - value: The runtime entity to get the meta representation of.
-meta : Any -> Atom | Constructor | Primitive | Polyglot | Unresolved_Symbol | Error
-meta value = if is_atom value then Atom.Value value else
-    if is_atom_constructor value then Constructor.Value value else
-        if is_polyglot value then Polyglot.Value value else
-            if is_unresolved_symbol value then Unresolved_Symbol.Value value else
-                if is_error value then Error.Value value.catch else
-                    if is_type value then Type.Value value.catch else
-                        Primitive.Value value
+meta ~value -> Atom | Constructor | Primitive | Polyglot | Unresolved_Symbol | Error | Type =
+    ac = Meta_Helpers.find_atom_constructor_builtin value...
+    kind = if ac.is_nothing.not then 1 else
+        Meta_Helpers.get_kind_builtin value
+
+    case kind:Integer of
+        1 -> Constructor.Value ac
+        2 -> Atom.Value value
```

Comment:
I expect _magic constant_ complains. However the most important thing is to keep these values consistent with the [builtin values](https://github.com/enso-org/enso/pull/12031/files#r1923159462). I don't think we have infrastructure for that. Just giving the numbers a name here and there won't help, imo....